CMakeLists.txt

```diff
@@ -11,12 +11,13 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-cmake_minimum_required(VERSION 3.0)
-project(aws-c-common C)
+cmake_minimum_required(VERSION 3.1)
+project(aws-c-common C CXX)
```

Comment:
It looks like adding CXX support to this project broke the clang 6 build. You may need to add `export CXX=XXX` to the codebuild scripts to ensure it's using the correct compiler for both languages.

You'll also want to add `-m32` to the CXX flags for the gcc-4x-x86 build script. Currently it's trying to build the c file in 32bit and the cxx files in 64bit, and then obviously the link fails.

Reply:
Xnay on adding CXX to this, this will enforce users having a C++ compiler to build.

It should be a separate project for this test (that is entirely optional) if we want to do this.